impress-org/givewp-next-gen
# Refactor: position form builder domain for scalability

## Description



As we move the form builder from prototype to MVP we need to position our domain for scalability.  This PR organizes FormBuilder's service provider with clearer defined WP actions that sets up our MVC workflow. 

The rest api requests have also been updated with extra validation and feature tests.   In the near future, the majority of this logic will come from a Form Model and the validation should continue to be improved.

## Affects



Nothing should actually be affected from a user perspective, just the underlying architecture of the Form Builder's domain.

## Visuals



<img width="1424" alt="Screen Shot 2022-09-15 at 9 25 42 AM" src="https://user-images.githubusercontent.com/10138447/190415621-61914792-b127-4855-aca2-28857c4d848d.png">

## Testing Instructions



Pull branch, run composer, run build, and use the form builder.  Everything should still work as expected.

## Pre-review Checklist
